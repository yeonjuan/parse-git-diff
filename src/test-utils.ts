import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
