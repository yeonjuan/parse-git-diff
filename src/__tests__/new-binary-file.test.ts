import { getFixture } from './test-utils';
import parseGitDiff from '../parse-git-diff';

describe('new-binary-file', () => {
  const fixture = getFixture('new-binary-file');

  it('parse `new-binary-file`', () => {
    expect(parseGitDiff(fixture)).toMatchSnapshot();
  });
});
