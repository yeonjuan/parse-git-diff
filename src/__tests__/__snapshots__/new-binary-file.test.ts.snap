// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new-binary-file parse \`new-binary-file\` 1`] = `
{
  "files": [
    {
      "chunks": [
        {
          "pathAfter": "file1",
          "pathBefore": "/dev/null",
          "type": "BinaryFilesChunk",
        },
      ],
      "path": "file1",
      "type": "AddedFile",
    },
  ],
  "type": "GitDiff",
}
`;
